src/pages/workspace/tags/WorkspaceViewTagsPage.tsx

```diff
@@ -316,11 +336,19 @@ function WorkspaceViewTagsPage({route}: WorkspaceViewTagsProps) {
                             title={translate('common.required')}
                             switchAccessibilityLabel={translate('common.required')}
                             isActive={!!currentPolicyTag?.required}
-                            onToggle={(on) => setPolicyTagsRequired(policyID, on, route.params.orderWeight)}
+                            onToggle={(on) => {
+                                if (getCountOfMultiLevelRequired(policyTags) === 1 && policy?.requiresTag && currentPolicyTag?.required) {
+                                    setShowCannotDisableLastTagModal(true);
+                                    return;
+                                }
+
+                                setPolicyTagsRequired(policyID, on, route.params.orderWeight);
+                            }}
                             pendingAction={currentPolicyTag.pendingFields?.required}
                             errors={currentPolicyTag?.errorFields?.required ?? undefined}
                             onCloseError={() => clearPolicyTagListErrorField(policyID, route.params.orderWeight, 'required')}
                             disabled={!currentPolicyTag?.required && !Object.values(currentPolicyTag?.tags ?? {}).some((tag) => tag.enabled)}
+                            showLockIcon={getCountOfMultiLevelRequired(policyTags) === 1 && policy?.requiresTag && currentPolicyTag?.required}
```

Comment:
`getCountOfMultiLevelRequired` is calculated more than once, move it to a variable